### PR TITLE
add extension icon and auto fill form

### PR DESCRIPTION
**Pull request by GingerBear**
@johnhof 
When I was working on katespade my account, I feel very annoying type login/password and address form again and again, and start missing the form filling feature of jj. 

The original one is seem just works for ember lite checkout, and little annoying for that on page button when it is not working properly. So I am trying to add a chrome extension button for jj. Currently it very simple. It just has a button to manually fill form and checkbox toggle auto fill form. and a very simple logo.
screen shot 2015-02-01 at 7 25 57 pm

It's very interesting that I never did this before. I found that one of the way extension view (just html) and page communicate is by injecting code. `runScriptInTab('jj.popForm();');`. This is used to make the Fill Form button works. The checkbox is using the `chrome.storage` to maintain state. 

Most checkout is rendering form after data fetched and currently no event firing. So the auto form filling is just keep trying to fill again and again, until get some fields updated or trying times limit reach. 

The mapping is trying to match name, if not then then match title. Form data is propagating to more fields names. 

I feel this may bring development a little convenience. Looking forward to your feedback John. :smirk:

### CODE REVIEW

**Comment by johnhof** [comment]
you are a one man army @GingerBear I'll try to get to this tom. I'll give you feedback and/or merge by tom night.

:+1: :100:  :gem:

**Comment by GingerBear** [comment]
cool, no rush.

**Comment by GingerBear** [comment]
oh, forget to mentions, the login/password is currently hardcoded to qatest. we may need to get ability to set then dynamically in the future.

**Comment by johnhof** [comment]
Yea, probably, but since this is an opt-in feature I wouldnt worry too much. This already does a lot more than the previous version. 

I'f you ever feel like it, youre welcome to gut the options page. I was using it to display cached jira boards, but I removed that functionality when we brought Jira in house. the option page doesnt do anything at the moment

**Comment by GingerBear** [comment]
I will take that.

**Comment by johnhof** [comment]
you dont have to do it in this PR. this is plenty functional. Just letting you know in case you were afraid to change it

File: content_scripts/utilities/form_populator/controller.js [hunk 1]
```diff
@@ -1,24 +1,70 @@
-// $('body').append('<div id="populator" class="jj-button">populate</div>');
+(function(window, $) {
+  window.jj = window.jj || {};
```
Comment (johnhof) [inline content_scripts/utilities/form_populator/controller.js]: very nice :+1:

**Comment by GingerBear** [comment]
yea, will do it in following pr

**Comment by johnhof** [comment]
No regression. I didnt test the actual functionality all that much. I'll prob make updates or give you feedback when I start using this for a site

**Comment by GingerBear** [comment]
sure thing

**Pull request merged**